app/lib/dartdoc/backend.dart

```diff
@@ -43,6 +46,60 @@ class DartdocBackend {
 
   DartdocBackend(this._db, this._storage);
 
+  /// Whether the storage bucket has a useable extracted data file.
+  /// Only the existence of the file is checked.
+  // TODO: decide whether we should re-generate the file after a certain age
+  Future<bool> hasValidDartSdkDartdocData() async {
+    final objectName =
+        storage_path.dartSdkDartdocDataName(shared_versions.runtimeVersion);
+    try {
+      final info = await _storage.info(objectName);
+      return info != null;
+    } catch (_) {
```

Comment:
Can we catch a specific exception instead?

Reply:
I recall two distinct kind of error there, but one of them may be related to the outage I have mentioned earlier.

Reply:
The other is `DetailedApiRequestError`, and I'm not sure where it is coming from.